### PR TITLE
feat(create-cloudflare): allow users to change prompt answer

### PR DESCRIPTION
**Pull request by edmundhung**
## What this PR solves / how to test

This PRs introduces a "Go back" option to most template related questions so users can go back to the previous question and change the answers.

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: No impact to wrangler
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: No changes in behavior.

### CODE REVIEW

File: packages/create-cloudflare/src/templates.ts [hunk 1]
```diff
@@ -159,8 +159,8 @@ export const getTemplateMap = async () => {
 	} as Record<string, TemplateConfig>;
 };
 
-export const inferRelatedArgs = (args: Partial<C3Args>) => {
-	// Infering the type based on the additional arguments provided
+export const deriveCorelatedArgs = (args: Partial<C3Args>) => {
```
Comment (Skye-31) [inline packages/create-cloudflare/src/templates.ts:162]: Little typo 🙂 ```suggestion
export const deriveCorrelatedArgs = (args: Partial<C3Args>) => {
```

Comment (edmundhung) [reply packages/create-cloudflare/src/templates.ts]: Oops! Thanks Skye.

File: packages/create-cloudflare/e2e-tests/cli.test.ts [hunk 2]
```diff
@@ -258,5 +258,98 @@ describe.skipIf(frameworkToTest || isQuarantineMode())(
...
+						},
+						{
+							matcher: /Which language do you want to use\?/,
+							input: {
+								type: "select",
+								target: "Go back",
+							},
+						},
+						{
+							matcher: /● Queue consumer & producer Worker/,
```
Comment (RamIdeas) [inline packages/create-cloudflare/e2e-tests/cli.test.ts:290]: Is this matcher not meant to match a prompt question? What does it mean to match an active option and then select "Go back"?

Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/cli.test.ts:290]: Yes. It is intentional to confirm that we have the previous selected option as the default value.

File: packages/create-cloudflare/src/templates.ts [hunk 3]
```diff
@@ -200,115 +202,249 @@ export const selectTemplate = async (args: Partial<C3Args>) => {
...
+				newArgs.framework = undefined;
+				args.framework = undefined;
+				break;
+			case "lang":
+				linesPrinted = 12;
+				newArgs.type = undefined;
+				newArgs.lang = undefined;
+				args.lang = undefined;
+				break;
+		}
```
Comment (RamIdeas) [inline packages/create-cloudflare/src/templates.ts:252]:
```suggestion
		switch (from) {
			case "type":
				linesPrinted = 9;
				newArgs.category = undefined;
				break;
			case "framework":
				linesPrinted = 9;
				newArgs.category = undefined;
				break;
			case "lang":
				linesPrinted = 12;
				newArgs.type = undefined;
				break;
		}
		
		newArgs[from] = undefined;
		args[from] = undefined;
```

File: packages/create-cloudflare/e2e-tests/cli.test.ts (hunk 2 above)
Comment (RamIdeas) [reply packages/create-cloudflare/e2e-tests/cli.test.ts:290]: Makes sense. But it's a bit implicit. If this tests fails in the future, I'm not sure the dev would know why this is the matcher.

We should either leave a comment on each of these matchers as-is or extend the PromptHandler config to match multiple regexes so we can test for the question _and_ this active element

Comment (edmundhung) [reply packages/create-cloudflare/e2e-tests/cli.test.ts]: I have updated the select prompt handler to support a `defaultValue` option.

**Pull request merged**